test/e2e/plugins_test.go

```diff
@@ -0,0 +1,115 @@
+package e2e
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	consolev1 "github.com/openshift/api/console/v1"
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	yaml "gopkg.in/yaml.v2"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/console/subresource/consoleserver"
+	"github.com/openshift/console-operator/test/e2e/framework"
+)
+
+const (
+	pluginName = "test-plugin"
+)
+
+func setupPluginsTestCase(t *testing.T) (*framework.ClientSet, *operatorsv1.Console) {
+	return framework.StandardSetup(t)
+}
+
+func cleanupPluginsTestCase(t *testing.T, client *framework.ClientSet) {
+	framework.StandardCleanup(t, client)
+
+	err := client.ConsolePlugin.Delete(context.TODO(), pluginName, metav1.DeleteOptions{})
+	if err != nil && !apiErrors.IsNotFound(err) {
+		t.Fatalf("could not delete cleanup %q plugin, %v", pluginName, err)
+	}
+	framework.StandardCleanup(t, client)
+}
+
+func TestCreatePlugin(t *testing.T) {
+	expectedPlugins := []string{pluginName}
+	currentPlugins := []string{}
+	client, _ := setupPluginsTestCase(t)
+	defer cleanupPluginsTestCase(t, client)
+
+	plugin := &consolev1.ConsolePlugin{
+		ObjectMeta: v1.ObjectMeta{
+			Name: pluginName,
+		},
+		Spec: consolev1.ConsolePluginSpec{
+			DisplayName: "TestPlugin",
+			Service: consolev1.ConsolePluginService{
+				Name:      "test-plugin-service-name",
+				Namespace: "test-plugin-service-namespace",
+				Port:      8443,
+				Manifest:  "/manifest",
```

Comment:
Please update this field to `BasePath` according to https://github.com/openshift/api/pull/764#discussion_r533645627